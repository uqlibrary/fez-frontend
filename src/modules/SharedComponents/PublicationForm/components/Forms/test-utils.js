import { useForm } from 'react-hook-form';
import { render, WithReduxStore } from 'test-utils';
import React from 'react';
import { formValues } from '../PublicationForm';

export const setupWrapper = (Form, props) => {
    const Wrapper = () => {
        const {
            control,
            formState: { isSubmitting },
        } = useForm();

        // eslint-disable-next-line react/prop-types
        return <Form {...{ control, isSubmitting, ...props, formValues: formValues(props?.formValues || {}) }} />;
    };

    return render(
        <WithReduxStore>
            <Wrapper />
        </WithReduxStore>,
    );
};
