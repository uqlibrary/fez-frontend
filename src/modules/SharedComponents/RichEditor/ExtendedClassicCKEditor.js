/**
 * based on https://github.com/jainsuneet/ckeditor5-build-classic-extended/blob/master/src/ckeditor.js
 */

// The editor creator to use.
import ClassicEditorBase from '@ckeditor/ckeditor5-editor-classic/src/classiceditor';

import Bold from '@ckeditor/ckeditor5-basic-styles/src/bold';
import Italic from '@ckeditor/ckeditor5-basic-styles/src/italic';
import Link from '@ckeditor/ckeditor5-link/src/link';

import RemoveFormat from '@ckeditor/ckeditor5-remove-format/src/removeformat';
import SpecialCharacters from '@ckeditor/ckeditor5-special-characters/src/specialcharacters';
import Strikethrough from '@ckeditor/ckeditor5-basic-styles/src/strikethrough';
import Subscript from '@ckeditor/ckeditor5-basic-styles/src/subscript';
import Superscript from '@ckeditor/ckeditor5-basic-styles/src/superscript';
import Underline from '@ckeditor/ckeditor5-basic-styles/src/underline';
import Capitalise from './components/Capitalise';

export default class ClassicEditor extends ClassicEditorBase {}

// Plugins to include in the build.
ClassicEditor.builtinPlugins = [
    Bold,
    Italic,
    Link,
    RemoveFormat,
    SpecialCharacters,
    Strikethrough,
    Subscript,
    Superscript,
    Underline,
];

ClassicEditor.plugins = [Capitalise];

// Editor configuration.
ClassicEditor.defaultConfig = {
    toolbar: {
        items: [
            'bold',
            'italic',
            'underline',
            'strikethrough',
            'subscript',
            'superscript',
            // ...(!!fieldProps.singleLine ? [] : ['|', 'link', 'numberedList', 'bulletedList']),
            '|',
            'removeFormat',
            'specialCharacters',
            'Capitalise',
            '|',
            'undo',
            'redo',
        ],
    },
    // This value must be kept in sync with the language defined in webpack.config.js.
    language: 'en',
};
