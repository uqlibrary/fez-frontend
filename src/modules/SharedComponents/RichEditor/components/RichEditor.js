import React from 'react';
import PropTypes from 'prop-types';

import { CKEditor } from '@ckeditor/ckeditor5-react';
import Editor from '../../../../../custom_modules/ckeditor5-custom-build';
import Typography from '@mui/material/Typography';

const RichEditor = fieldProps => {
    const editorConfig = {
        toolbar: {
            items: [
                'bold',
                'italic',
                'underline',
                'strikethrough',
                'subscript',
                'superscript',
                ...(!!fieldProps.singleLine ? [] : ['|', 'link', 'numberedList', 'bulletedList']),
                '|',
                'LetterCase',
                'removeFormat',
                'specialCharacters',
                '|',
                'undo',
                'redo',
            ],
        },
        removePlugins: ['MediaEmbedToolbar'],
    };

    function handleEditorDataChange(event, editor) {
        const textValue = editor.getData().replace(/<[^>]*>?/gm, '');
        const newTypedValue =
            textValue.length > 0
                ? {
                      htmlText: editor.getData(),
                      plainText: textValue,
                  }
                : null;
        fieldProps.onChange(newTypedValue);
    }

    function getContent() {
        let dataForEditor = '';
        /* istanbul ignore else */
        if (fieldProps?.input?.value?.size > 0) {
            dataForEditor = fieldProps.input.value.get('htmlText') || fieldProps.input.value.get('plainText') || '';
        } else if (!!fieldProps && fieldProps.hasOwnProperty('value')) {
            if (!!fieldProps.value.get && !!fieldProps.value.get('htmlText')) {
                dataForEditor = fieldProps.value.get('htmlText');
            } else if (!!fieldProps.value.htmlText) {
                dataForEditor = fieldProps.value.htmlText;
            } else if (typeof fieldProps.value === 'string' && fieldProps.value.length > 0) {
                dataForEditor = fieldProps.value;
            }
        }
        return typeof dataForEditor === 'string' ? dataForEditor : /* istanbul ignore next */ '';
    }

    let error = null;
    const inputLength =
        (fieldProps.value && fieldProps.value.plainText && fieldProps.value.plainText.length) ||
        fieldProps.value.length - 7; // default rich editor has "<p></p>"
    if (fieldProps.meta && fieldProps.meta.error) {
        error =
            !!fieldProps.meta.error.props &&
            React.Children.map(fieldProps.meta.error.props.children, (child, index) => {
                if (child.type) {
                    return React.cloneElement(child, {
                        key: index,
                    });
                } else {
                    return child;
                }
            });
    }
    // rendered content of empty CKEditor:
    // <p><br data-cke-filler="true"></p>
    return (
        <div
            id={fieldProps.richEditorId}
            data-testid={fieldProps.richEditorId}
            data-analyticsid={fieldProps.richEditorId}
        >
            <span>
                {fieldProps.title && (
                    <Typography color={fieldProps.meta && fieldProps.meta.error && 'error'} {...fieldProps.titleProps}>
                        {fieldProps.title}
                        {fieldProps.required && <span> *</span>}
                    </Typography>
                )}
                {fieldProps.description && (
                    <Typography color={fieldProps.meta && fieldProps.meta.error && 'error'} variant={'caption'}>
                        {fieldProps.description}
                    </Typography>
                )}
            </span>
            <CKEditor
                className={fieldProps.className}
                editor={Editor}
                config={editorConfig}
                data={getContent()}
                onReady={editor => {
                    const documentView = editor.editing.view.document;
                    /* istanbul ignore next */
                    documentView.on('clipboardInput', (event, data) => {
                        data.content = editor.data.htmlProcessor.toView(data.dataTransfer.getData('text/plain'));
                    });
                }}
                onChange={(event, editor) => {
                    handleEditorDataChange(event, editor);
                }}
            />
            {fieldProps.meta && fieldProps.meta.error && (
                <Typography
                    color="error"
                    variant="caption"
                    component={'span'}
                    style={{
                        display: 'inline-block',
                    }}
                >
                    {error || fieldProps.meta.error}
                    {fieldProps.maxValue && <span>&nbsp;-&nbsp;</span>}
                </Typography>
            )}
            {fieldProps.maxValue && (
                <Typography
                    component={'span'}
                    style={{
                        display: 'inline-block',
                    }}
                    variant="caption"
                    color={fieldProps.meta && fieldProps.meta.error && 'error'}
                >
                    {inputLength > 0 ? inputLength : 0} characters of {fieldProps.maxValue}
                    {fieldProps.instructions || ''}
                </Typography>
            )}
        </div>
    );
};

RichEditor.prototypes = {
    className: PropTypes.string,
    input: PropTypes.object,
    inputRef: PropTypes.any,
    instructions: PropTypes.any,
    maxValue: PropTypes.number,
    meta: PropTypes.any,
    onChange: PropTypes.func.isRequired,
    richEditorId: PropTypes.string,
    required: PropTypes.bool,
    singleLine: PropTypes.bool,
    title: PropTypes.string,
    value: PropTypes.string,
};

RichEditor.defaultProps = {
    className: '',
    disabled: false,
    required: false,
    singleLine: false,
    value: '',
};

export default RichEditor;
