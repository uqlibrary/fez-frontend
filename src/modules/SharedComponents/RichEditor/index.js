export { default as RichEditorField } from './RichEditorField';
