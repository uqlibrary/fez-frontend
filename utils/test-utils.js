/* eslint-disable react/prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { createMemoryRouter, RouterProvider } from 'react-router-dom';
import { mui1theme } from 'config/theme';
import { Provider } from 'react-redux';

import { ThemeProvider as MuiThemeProvider } from '@mui/material/styles';
import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
import { AdapterMoment } from '@mui/x-date-pickers/AdapterMoment';

import { getStore } from '../src/config/store';
import Immutable from 'immutable';

import mediaQuery from 'css-mediaquery';

const domTestingLib = require('@testing-library/dom');
const reactTestingLib = require('@testing-library/react');
const mime = require('mime-types');

import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { waitFor } from '@testing-library/dom';
import preview from 'jest-preview';

export const AllTheProviders = props => {
    return (
        <MuiThemeProvider theme={mui1theme}>
            <LocalizationProvider dateAdapter={AdapterMoment}>{props.children}</LocalizationProvider>
        </MuiThemeProvider>
    );
};

AllTheProviders.propTypes = {
    children: PropTypes.node,
};

export const rtlRender = (ui, options) => {
    return {
        user: userEvent.setup(),
        ...render(ui, { wrapper: AllTheProviders, ...options }),
    };
};

export const WithRouter = ({ children, route = '/', initialEntries = [route] }) => {
    const routes = [{ path: route, element: children }];
    const router = createMemoryRouter(routes, { initialEntries: initialEntries });
    return <RouterProvider router={router} />;
};

export const withRedux = (initialState = Immutable.Map()) => WrappedComponent => {
    return <Provider store={getStore(initialState)}>{WrappedComponent}</Provider>;
};

export const WithReduxStore = ({ initialState = Immutable.Map(), children }) => (
    <Provider store={getStore(initialState)}>
        <AllTheProviders>{children}</AllTheProviders>
    </Provider>
);

export const assertTooltipText = async (trigger, tooltipText) => {
    expect(trigger).toBeInTheDocument();
    await userEvent.hover(trigger);
    await waitFor(() => {
        expect(screen.getByRole('tooltip', { name: tooltipText, hidden: true })).toBeVisible();
    });
};

const extensionToMimeMap = {
    tiff: 'image/tiff',
    wav: 'audio/wave',
    zip: 'application/zip',
};
const mimeType = filename => extensionToMimeMap[filename.split('.').pop()];

export const defaultDatastream = {
    dsi_embargo_date: null,
    dsi_open_access: 1,
    dsi_label: '',
    dsi_copyright: null,
    dsi_state: 'A',
    dsi_size: 27932352,
};

export const createFezDatastreamInfoArray = (datastreams, pid = null, withPreview = true) => {
    let processed = datastreams.filter(datastream => typeof datastream === 'object' || typeof datastream === 'string');
    if (withPreview) {
        processed = [];
        datastreams.forEach(datastream => {
            processed.push(datastream);
            const filename = typeof datastream === 'object' ? datastream.dsi_dsid : datastream;
            const mimetype = mimeType(filename);
            // bail in case it's not derivable
            if (
                !filename.includes('.pdf') &&
                !mimetype.includes('image') &&
                !mimetype.includes('audio') &&
                !mimetype.includes('video')
            ) {
                return;
            }
            // eslint-disable-next-line no-nested-ternary
            const derivativeMimetype = mimetype.includes('audio')
                ? 'audio/mp3'
                : mimetype.includes('video')
                ? 'video/mp4'
                : 'image/jpg';

            (derivativeMimetype.includes('image') ? ['preview_', 'thumbnail_', 'web_'] : ['']).forEach(prefix => {
                processed.push({
                    dsi_dsid: `${prefix}${filename.split('.')[0]}_t.${derivativeMimetype.split('/').pop()}`,
                    dsi_mimetype: derivativeMimetype,
                });
            });
        });
    }

    return processed.map((datastream, index) => ({
        ...defaultDatastream,
        ...{
            dsi_id: index + 1,
            dsi_pid: pid,
            dsi_order: index + 1,
            dsi_mimetype: mime.lookup(typeof datastream === 'object' ? datastream.dsi_dsid : datastream),
        },
        ...(typeof datastream === 'object'
            ? datastream
            : {
                  dsi_dsid: datastream,
              }),
    }));
};

export const withDatastreams = (sources, datastreams, callback) =>
    datastreams.map(item => {
        const source = sources.find(datastream =>
            datastream.dsi_dsid.includes(
                item.dsi_dsid
                    .replace('_t.', '.')
                    .replace('preview_', '')
                    .replace('thumbnail_', '')
                    .replace('web_', '')
                    .split('.')[0],
            ),
        );
        const isDerivative = item.dsi_dsid !== source.dsi_dsid;
        callback(item, source, isDerivative);
    });

export const getDatastreamByFilename = (filename, datastreams) =>
    datastreams.find(datastream => datastream.dsi_dsid === filename);

export const createMatchMedia = width => {
    return query => ({
        matches: mediaQuery.match(query, { width }),
        /* istanbul ignore next */
        addListener: () => {},
        /* istanbul ignore next */
        removeListener: () => {},
    });
};

export const getFilenameExtension = filename => filename.split('.').pop();
export const getFilenameBasename = filename =>
    filename.replace(new RegExp(`/\.${getFilenameExtension(filename)}$/`), '');
export const addFilesToFileUploader = files => {
    const { screen, fireEvent } = reactTestingLib;
    // create a list of Files
    const fileList = files.map(file => {
        // if file it's a string, treat it as a filename
        if (typeof file === 'string') {
            return new File([getFilenameBasename(file)], file, { type: `image/${getFilenameExtension(file)}` });
        }
        // otherwise expect it to be a object with filename and mimeType keys
        return new File([getFilenameBasename(file.filename)], file.filename, { type: file.mimeType });
    });
    // drag and drop files
    fireEvent.drop(screen.getByTestId('fez-datastream-info-input'), {
        dataTransfer: {
            files: fileList,
            types: ['Files'],
        },
    });
};
export const setFileUploaderFilesToClosedAccess = async (files, timeout = 500) => {
    const { fireEvent } = reactTestingLib;
    // set all files to closed access
    for (const file of files) {
        const index = files.indexOf(file);
        await waitFor(() => screen.getByText(new RegExp(getFilenameBasename(file))), { timeout });
        fireEvent.mouseDown(screen.getByTestId(`dsi-open-access-${index}-select`));
        fireEvent.click(screen.getByRole('option', { name: 'Closed Access' }));
    }
};

export const assertEnabled = (resolver, ...items) => items.forEach(item => expect(resolver(item)).toBeEnabled());
export const assertDisabled = (resolver, ...items) => items.forEach(item => expect(resolver(item)).toBeDisabled());

module.exports = {
    ...domTestingLib,
    ...reactTestingLib,
    rtlRender,
    withRedux,
    AllTheProviders,
    WithReduxStore,
    assertTooltipText,
    WithRouter,
    createFezDatastreamInfoArray,
    getDatastreamByFilename,
    withDatastreams,
    createMatchMedia,
    preview,
    userEvent,
    assertEnabled,
    assertDisabled,
    getFilenameExtension,
    getFilenameBasename,
    addFilesToFileUploader,
    setFileUploaderFilesToClosedAccess,
};
