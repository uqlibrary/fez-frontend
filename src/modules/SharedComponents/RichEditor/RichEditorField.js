import React from 'react';
import RichEditor from './components/RichEditor';

export default function RichEditorField(fieldProps) {
    return (
        <RichEditor
            onChange={fieldProps?.input?.onChange ?? fieldProps?.onChange}
            value={fieldProps.input?.value ?? fieldProps.value}
            {...fieldProps}
        />
    );
}
