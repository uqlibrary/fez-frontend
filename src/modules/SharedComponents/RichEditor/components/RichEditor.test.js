import React from 'react';
import RichEditor from './RichEditor';
import Immutable from 'immutable';
import { rtlRender } from 'test-utils';

class ResizeObserver {
    observe() {}
    unobserve() {}
    disconnect() {}
}

window.ResizeObserver = ResizeObserver;

function setup(testProps = {}, renderMethod = rtlRender) {
    const props = {
        onChange: jest.fn(), // PropTypes.func.isRequired,
        disabled: false,
        richEditorId: 'rek-test',
        inputRef: {
            current: <div />,
        },
        ...testProps,
    };

    return renderMethod(<RichEditor {...props} />);
}

describe('RichEditor', () => {
    const setReadOnlyFn = jest.fn();
    const setDataFn = jest.fn();

    afterEach(() => {
        setReadOnlyFn.mockClear();
        setDataFn.mockClear();
    });

    it('should render component', () => {
        const { container } = setup({});
        expect(container).toMatchSnapshot();
    });

    it('should render given title and description', () => {
        const { container } = setup({
            title: 'This is test title',
            description: 'This is test description',
        });
        expect(container).toMatchSnapshot();
    });

    it('should render given title and description with error', () => {
        const { container } = setup({
            title: 'This is title with error',
            description: 'This is description with error',
            meta: {
                error: 'This field is required',
            },
            required: true,
        });
        expect(container).toMatchSnapshot();
    });

    it('should render error showing maxValue and instructions', () => {
        const { container } = setup({
            value: Immutable.Map({ htmlText: 'This is test value' }),
            maxValue: 10,
            instructions: 'test instructions',
            meta: {
                error: 'This field is required',
            },
            required: true,
        });
        expect(container).toMatchSnapshot();
    });

    it('should render maxValue with missing instructions', () => {
        const { container } = setup({
            maxValue: 10,
            required: true,
        });
        expect(container).toMatchSnapshot();
    });

    it('should render error showing input length', () => {
        const { container } = setup({
            value: { plainText: 'This is test value', get: jest.fn() },
            maxValue: 10,
            instructions: 'test instructions',
            meta: {
                error: 'This field is required',
            },
        });
        expect(container).toMatchSnapshot();
    });

    it('should render error as react children', () => {
        const { container } = setup({
            value: { plainText: 'This is test value', get: jest.fn() },
            maxValue: 10,
            instructions: 'test instructions',
            meta: {
                error: (
                    <p>
                        <span>This field is required</span>
                    </p>
                ),
            },
        });
        expect(container).toMatchSnapshot();
    });

    it('should render error as one child', () => {
        const { container } = setup({
            value: { plainText: 'This is test value', get: jest.fn() },
            maxValue: 10,
            instructions: 'test instructions',
            meta: {
                error: <span>This field is required</span>,
            },
        });
        expect(container).toMatchSnapshot();
    });

    it('should render plain text value', () => {
        const { container } = setup({ value: 'test' });
        expect(container).toMatchSnapshot();
    });

    it('should handle null value', () => {
        const { container } = setup({ value: null });
        expect(container).toMatchSnapshot();
    });
});
