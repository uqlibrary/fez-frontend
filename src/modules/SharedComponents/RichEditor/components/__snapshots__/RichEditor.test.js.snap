// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RichEditor should handle null value 1`] = `
<div>
  <div
    data-analyticsid="rek-test"
    data-testid="rek-test"
    id="rek-test"
  >
    <span />
    <div />
  </div>
</div>
`;

exports[`RichEditor should render component 1`] = `
<div>
  <div
    data-analyticsid="rek-test"
    data-testid="rek-test"
    id="rek-test"
  >
    <span />
    <div />
  </div>
</div>
`;

exports[`RichEditor should render error as one child 1`] = `
<div>
  <div
    data-analyticsid="rek-test"
    data-testid="rek-test"
    id="rek-test"
  >
    <span />
    <div />
    <span
      class="MuiTypography-root MuiTypography-caption css-1xktwm9-MuiTypography-root"
      style="display: inline-block;"
    >
      This field is required
      <span>
         - 
      </span>
    </span>
    <span
      class="MuiTypography-root MuiTypography-caption css-1xktwm9-MuiTypography-root"
      style="display: inline-block;"
    >
      18
       characters of 
      10
      test instructions
    </span>
  </div>
</div>
`;

exports[`RichEditor should render error as react children 1`] = `
<div>
  <div
    data-analyticsid="rek-test"
    data-testid="rek-test"
    id="rek-test"
  >
    <span />
    <div />
    <span
      class="MuiTypography-root MuiTypography-caption css-1xktwm9-MuiTypography-root"
      style="display: inline-block;"
    >
      <span>
        This field is required
      </span>
      <span>
         - 
      </span>
    </span>
    <span
      class="MuiTypography-root MuiTypography-caption css-1xktwm9-MuiTypography-root"
      style="display: inline-block;"
    >
      18
       characters of 
      10
      test instructions
    </span>
  </div>
</div>
`;

exports[`RichEditor should render error showing input length 1`] = `
<div>
  <div
    data-analyticsid="rek-test"
    data-testid="rek-test"
    id="rek-test"
  >
    <span />
    <div />
    <span
      class="MuiTypography-root MuiTypography-caption css-1xktwm9-MuiTypography-root"
      style="display: inline-block;"
    >
      This field is required
      <span>
         - 
      </span>
    </span>
    <span
      class="MuiTypography-root MuiTypography-caption css-1xktwm9-MuiTypography-root"
      style="display: inline-block;"
    >
      18
       characters of 
      10
      test instructions
    </span>
  </div>
</div>
`;

exports[`RichEditor should render error showing maxValue and instructions 1`] = `
<div>
  <div
    data-analyticsid="rek-test"
    data-testid="rek-test"
    id="rek-test"
  >
    <span />
    <div />
    <span
      class="MuiTypography-root MuiTypography-caption css-1xktwm9-MuiTypography-root"
      style="display: inline-block;"
    >
      This field is required
      <span>
         - 
      </span>
    </span>
    <span
      class="MuiTypography-root MuiTypography-caption css-1xktwm9-MuiTypography-root"
      style="display: inline-block;"
    >
      0
       characters of 
      10
      test instructions
    </span>
  </div>
</div>
`;

exports[`RichEditor should render given title and description 1`] = `
<div>
  <div
    data-analyticsid="rek-test"
    data-testid="rek-test"
    id="rek-test"
  >
    <span>
      <p
        class="MuiTypography-root MuiTypography-body1 css-u1nwny-MuiTypography-root"
      >
        This is test title
      </p>
      <span
        class="MuiTypography-root MuiTypography-caption css-1e4ju4k-MuiTypography-root"
      >
        This is test description
      </span>
    </span>
    <div />
  </div>
</div>
`;

exports[`RichEditor should render given title and description with error 1`] = `
<div>
  <div
    data-analyticsid="rek-test"
    data-testid="rek-test"
    id="rek-test"
  >
    <span>
      <p
        class="MuiTypography-root MuiTypography-body1 css-v0lc22-MuiTypography-root"
      >
        This is title with error
        <span>
           *
        </span>
      </p>
      <span
        class="MuiTypography-root MuiTypography-caption css-1xktwm9-MuiTypography-root"
      >
        This is description with error
      </span>
    </span>
    <div />
    <span
      class="MuiTypography-root MuiTypography-caption css-1xktwm9-MuiTypography-root"
      style="display: inline-block;"
    >
      This field is required
    </span>
  </div>
</div>
`;

exports[`RichEditor should render maxValue with missing instructions 1`] = `
<div>
  <div
    data-analyticsid="rek-test"
    data-testid="rek-test"
    id="rek-test"
  >
    <span />
    <div />
    <span
      class="MuiTypography-root MuiTypography-caption css-1e4ju4k-MuiTypography-root"
      style="display: inline-block;"
    >
      0
       characters of 
      10
    </span>
  </div>
</div>
`;

exports[`RichEditor should render plain text value 1`] = `
<div>
  <div
    data-analyticsid="rek-test"
    data-testid="rek-test"
    id="rek-test"
  >
    <span />
    <div />
  </div>
</div>
`;
